.github/workflows/main.yml

```diff
@@ -19,17 +19,48 @@ jobs:
         python-version: ['3.9', '3.10']
     timeout-minutes: 20
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v4
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install .
-        pip install -r requirements/test-requirements.txt
-    - name: Run tests
-      run: |
-        pytest tests/ai/
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install .
+          pip install -r requirements/test-requirements.txt
+      - name: Run tests
+        run: |
+          # test pydantic v1
+          pip install 'pydantic<2'
+          pytest tests/ai/
+          # test pydantic v2
+          pip install 'pydantic>=2'
```

Comment:
I'd drop support for v1. Supporting both is too painful.

Reply:
But langchain still uses pydantic v1, would it prevent users from using our lib because of this?

Reply:
langchain doesn't support v2?

Reply:
<img width="984" alt="image" src="https://github.com/user-attachments/assets/ae81e99e-9a33-411a-a504-e123e1c8a9db">
https://python.langchain.com/v0.2/docs/how_to/pydantic_compatibility/#1-passing-pydantic-objects-to-langchain-apis

Reply:
damn